packtools/sps/formats/crossref.py

```diff
@@ -1224,9 +1224,8 @@ def xml_crossref_articlecontributors_pipe(xml_crossref, xml_tree):
     """
     articles = article_and_subarticles.ArticleAndSubArticles(xml_tree)
     article_data = articles.data
-    article_nodes = articles.article
-    for article_node in article_nodes:
-        authors = list(article_authors.Authors(article_node).contribs_with_affs)
+
+    authors = [author.contrib_with_aff for author in article_authors.Authors(xml_tree).article_contribs]
```

Comment:
@Rossi-Luciano aqui está gerando uma lista sem necessidade, pois será usada mais adiante dentro de um for. Ou seja, está executando 2x o mesmo for. No lugar de iterar sobre a lista authors, simplesmente use a instrução do 'for' desta linha no lugar da linha 1230, com os devidos ajustes